### PR TITLE
Fixed bug that showed opaque window when solar reflectance was zero.

### PR DESCRIPTION
**Pull request by JasonGlazer**
Addresses #4580

### CODE REVIEW

**Comment by JasonGlazer** [comment]
@vidanovic I am just copying the comment on the issue over to the pull request.

This looks like it solves most of the problems. I just did some testing of the original bug file and I think this addresses

```
Surface Window Transmitted Solar Radiation Rate W
Surface Window Transmitted Beam Solar Radiation Rate W
Surface Window Transmitted Beam To Beam Solar Radiation Rate W
Surface Window Transmitted Beam To Diffuse Solar Radiation Rate W
Surface Window Transmitted Diffuse Solar Radiation Rate W
Surface Window Heat Gain Rate W
```

But now the values of do not seem to be quite correct:

```
Surface Window Total Glazing Layers Absorbed Solar Radiation Rate W
Surface Window Total Glazing Layers Absorbed Shortwave Radiation Rate W
Surface Window Total Absorbed Shortwave Radiation Rate Layer 1 W
```

They are not zero but very close (usually on the order of 1e-12). The test file without the bug shows these values range from 0.2 to 8.5.

**Comment by nrel-bot-3** [comment]
@JasonGlazer @lgentile it has been 14 days since this pull request was last updated.

File: src/EnergyPlus/WindowManager.cc [hunk 1]
```diff
@@ -5358,13 +5358,13 @@ namespace WindowManager {
 				tpb1 = 1 - rpf1;
 				tpb2 = 1 - rpf2;
 				tmp7 = -abf;
-				if ( tmp7 != 0.0 ) {
+				if ( cgf != 0.0 ) {
 					expmabfdivcgf = std::exp( tmp7 / cgf );
```
Comment (Myoldmopar) [inline src/EnergyPlus/WindowManager.cc:5520]: `expmabfdivcgf`

Umm...

``` c++
#include <iostream>
#include <string>
#include <ctime>
int main(){
 std::string crazyVarName = "";
 unsigned seed = time(0);
 srand(seed);
 for (int i = 0; i <= 12; i++) {
  crazyVarName += 'a' + rand()%26;
 }
 std::cout << "Crazy variable name = " << crazyVarName << std::endl;
}
// Sample output: Crazy variable name = jxtzoivhtnoox
```

**Comment by vidanovic** [comment]
:)
Actually expmabdfdivsgf is exactly what is written on the right side. Variable describes the entire equation :)

**Comment by nrel-bot-3** [comment]
@JasonGlazer @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@JasonGlazer @lgentile it has been 14 days since this pull request was last updated.

**Comment by mjwitte** [comment]
@JasonGlazer I've merged in develop and re-ran the defect files.  Per your earlier [comments](https://github.com/NREL/EnergyPlus/pull/5312#issuecomment-155494916) the window transmitted solar has reasonable results now.  The absorbed radiation values should be tiny since the transmittance is 0.999, and I see values on the order of <1e-10 in both files.  The only problem I see is that there are now some negative values for absorbed radiation.  Even though they are tiny values, it would be good to trap that.

**Comment by JasonGlazer** [comment]
@mjwitte If you run the file without the bug, you will see that those other variables should be showing values that are from 0.2 to 8.5. Those are the values I was trying to match.

**Comment by JasonGlazer** [comment]
@mjwitte The change to HeatBalanceSurfaceManager.cc to make sure the QRadSWwinAbs is not negative for the case used in the test file ends up causing a huge number of diffs. I am inclined to revert back to the old code which should not any diffs. What do you think?

**Comment by mjwitte** [comment]
@JasonGlazer That's a bit disconcerting.  Have you looked at one or two of the files that change to see if they are reporting negative window absorption values without this change?

**Comment by JasonGlazer** [comment]
Yes, I added the output variables to the RefBldgSmallOfficeNew2004_Chicago.idf and it shows some negative absorption values. See the spreadsheet. Columns E to BO are the new variables and rows below 50 show the negative numbers.

[30.zip](https://github.com/NREL/EnergyPlus/files/200171/30.zip)

**Comment by mjwitte** [comment]
I see, and they are not all tiny, some as large as -34W.  So, I would post a new issue for the negative absorbed radiation problem and remove the negative check from here.  Then this PR can be finished with just the reflectance fix and the negative number problem can be investigated separately.

**Comment by JasonGlazer** [comment]
@mjwitte and @Myoldmopar all the tests are passing now that we are not eliminating negative absorption. Could one of you please review this? I think it is ready to be merged.

**Comment by Myoldmopar** [comment]
I have no concerns with this change.  @vidanovic or @mjwitte any final thoughts?

**Comment by vidanovic** [comment]
@Myoldmopar I think separate issue is ok.  I did try couple of months ago to take a look into absorptance issue and realized that it will need more than a day to take a look into equations.

**Pull request merged**